### ISSUE TITLE
代码问题

### ISSUE THREAD

**Issue by ShifaTang** [xbchen82/PedCMT#2]
我用提供的代码，用JAAD数据集跑出来的结果，跟实际完全不符
结果如下
Acc: 0.6394557823129252
 f1: 0.7800829875518672
 precision_score: 0.6394557823129252
 recall_score: 1.0
 roc_auc_score: 0.5
 confusion_matrix: [[ 0 53]
 [ 0 94]]

Acc: 0.18076923076923077
 f1: 0.3061889250814332
 precision_score: 0.18076923076923077
 recall_score: 1.0
 roc_auc_score: 0.5
 confusion_matrix: [[  0 426]
 [  0  94]]

这表明模型全都把类别只分成负类，模型没起效果

**Comment by xbchen82** [comment]
这是库的版本和跑出的实验结果

name: pedCMT
channels:
  - pytorch
  - defaults
  - conda-forge
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=5.1=1_gnu
  - absl-py=2.1.0=py38h06a4308_0
  - aiohappyeyeballs=2.4.0=py38h06a4308_0
  - aiohttp=3.10.5=py38h5eee18b_0
  - aiosignal=1.2.0=pyhd3eb1b0_0
  - async-timeout=4.0.3=py38h06a4308_0
  - attrs=24.2.0=py38h06a4308_0
  - blas=1.0=mkl
  - blinker=1.6.2=py38h06a4308_0
  - brotli-python=1.0.9=py38h6a678d5_8
  - c-ares=1.19.1=h5eee18b_0
  - ca-certificates=2024.9.24=h06a4308_0
  - cachetools=5.3.3=py38h06a4308_0
  - certifi=2024.8.30=py38h06a4308_0
  - cffi=1.17.1=py38h1fdaa30_0
  - charset-normalizer=3.3.2=pyhd3eb1b0_0
  - click=8.1.7=py38h06a4308_0
  - cryptography=43.0.0=py38hdda0065_0
  - cudatoolkit=11.3.1=h2bc3f7f_2
  - frozenlist=1.4.0=py38h5eee18b_0
  - future=0.18.3=py38h06a4308_0
  - google-auth=2.29.0=py38h06a4308_0
  - google-auth-oauthlib=0.4.1=py_2
  - grpcio=1.62.2=py38h6a678d5_0
  - idna=3.7=py38h06a4308_0
  - importlib-metadata=7.0.1=py38h06a4308_0
  - intel-openmp=2021.4.0=h06a4308_3561
  - ld_impl_linux-64=2.40=h12ee557_0
  - libabseil=20240116.2=cxx17_h6a678d5_0
  - libffi=3.4.4=h6a678d5_1
  - libgcc-ng=11.2.0=h1234567_1
  - libgomp=11.2.0=h1234567_1
  - libgrpc=1.62.2=h2d74bed_0
  - libprotobuf=4.25.3=he621ea3_0
  - libstdcxx-ng=11.2.0=h1234567_1
  - libuv=1.43.0=h7f98852_0
  - markdown=3.4.1=py38h06a4308_0
  - markupsafe=2.1.3=py38h5eee18b_0
  - mkl=2021.4.0=h06a4308_640
  - mkl-service=2.4.0=py38h95df7f1_0
  - mkl_fft=1.3.1=py38h8666266_1
  - mkl_random=1.2.2=py38h1abd341_0
  - multidict=6.0.4=py38h5eee18b_0
  - ncurses=6.4=h6a678d5_0
  - ninja=1.11.0=h924138e_0
  - numpy=1.24.3=py38h14f4228_0
  - numpy-base=1.24.3=py38h31eccc5_0
  - oauthlib=3.2.2=py38h06a4308_0
  - openssl=3.0.15=h5eee18b_0
  - pip=24.2=py38h06a4308_0
  - protobuf=4.25.3=py38h12ddb61_0
  - pyasn1=0.4.8=pyhd3eb1b0_0
  - pyasn1-modules=0.2.8=py_0
  - pycparser=2.21=pyhd3eb1b0_0
  - pyjwt=2.9.0=pyhd8ed1ab_1
  - pyopenssl=24.2.1=py38h06a4308_0
  - pysocks=1.7.1=py38h06a4308_0
  - python=3.8.20=he870216_0
  - python_abi=3.8=2_cp38
  - pytorch=1.11.0=py3.8_cuda11.3_cudnn8.2.0_0
  - pytorch-mutex=1.0=cuda
  - pyyaml=6.0.2=py38h5eee18b_0
  - re2=2022.04.01=h295c915_0
  - readline=8.2=h5eee18b_0
  - requests=2.32.3=py38h06a4308_0
  - requests-oauthlib=2.0.0=py38h06a4308_0
  - rsa=4.7.2=pyhd3eb1b0_1
  - setuptools=75.1.0=py38h06a4308_0
  - six=1.16.0=pyh6c4a22f_0
  - sqlite=3.45.3=h5eee18b_0
  - tensorboard=2.6.0=py_0
  - tensorboard-plugin-wit=1.8.1=py38h06a4308_0
  - tk=8.6.14=h39e8969_0
  - typing-extensions=4.12.2=hd8ed1ab_0
  - typing_extensions=4.12.2=pyha770c72_0
  - urllib3=2.2.3=py38h06a4308_0
  - werkzeug=3.0.3=py38h06a4308_0
  - wheel=0.44.0=py38h06a4308_0
  - xz=5.4.6=h5eee18b_1
  - yaml=0.2.5=h7b6447c_0
  - yarl=1.11.0=py38h5eee18b_0
  - zipp=3.20.2=py38h06a4308_0
  - zlib=1.2.13=h5eee18b_1
  - pip:
      - einops==0.8.0
      - joblib==1.4.2
      - opencv-python==4.10.0.84
      - pickle5==0.0.11
      - scikit-learn==1.3.2
      - scipy==1.10.1
      - threadpoolctl==3.5.0
      - tqdm==4.67.0
prefix: /home/ubuntu/anaconda3/envs/pedCMT


[results.txt](https://github.com/user-attachments/files/17760689/results.txt)

**Comment by ShifaTang** [comment]
> 这是库的版本和跑出的实验结果
> 
> name: pedCMT channels:
> 
> * pytorch
> * defaults
> * conda-forge
>   dependencies:
> * _libgcc_mutex=0.1=main
> * _openmp_mutex=5.1=1_gnu
> * absl-py=2.1.0=py38h06a4308_0
> * aiohappyeyeballs=2.4.0=py38h06a4308_0
> * aiohttp=3.10.5=py38h5eee18b_0
> * aiosignal=1.2.0=pyhd3eb1b0_0
> * async-timeout=4.0.3=py38h06a4308_0
> * attrs=24.2.0=py38h06a4308_0
> * blas=1.0=mkl
> * blinker=1.6.2=py38h06a4308_0
> * brotli-python=1.0.9=py38h6a678d5_8
> * c-ares=1.19.1=h5eee18b_0
> * ca-certificates=2024.9.24=h06a4308_0
> * cachetools=5.3.3=py38h06a4308_0
> * certifi=2024.8.30=py38h06a4308_0
> * cffi=1.17.1=py38h1fdaa30_0
> * charset-normalizer=3.3.2=pyhd3eb1b0_0
> * click=8.1.7=py38h06a4308_0
> * cryptography=43.0.0=py38hdda0065_0
> * cudatoolkit=11.3.1=h2bc3f7f_2
> * frozenlist=1.4.0=py38h5eee18b_0
> * future=0.18.3=py38h06a4308_0
> * google-auth=2.29.0=py38h06a4308_0
> * google-auth-oauthlib=0.4.1=py_2
> * grpcio=1.62.2=py38h6a678d5_0
> * idna=3.7=py38h06a4308_0
> * importlib-metadata=7.0.1=py38h06a4308_0
> * intel-openmp=2021.4.0=h06a4308_3561
> * ld_impl_linux-64=2.40=h12ee557_0
> * libabseil=20240116.2=cxx17_h6a678d5_0
> * libffi=3.4.4=h6a678d5_1
> * libgcc-ng=11.2.0=h1234567_1
> * libgomp=11.2.0=h1234567_1
> * libgrpc=1.62.2=h2d74bed_0
> * libprotobuf=4.25.3=he621ea3_0
> * libstdcxx-ng=11.2.0=h1234567_1
> * libuv=1.43.0=h7f98852_0
> * markdown=3.4.1=py38h06a4308_0
> * markupsafe=2.1.3=py38h5eee18b_0
> * mkl=2021.4.0=h06a4308_640
> * mkl-service=2.4.0=py38h95df7f1_0
> * mkl_fft=1.3.1=py38h8666266_1
> * mkl_random=1.2.2=py38h1abd341_0
> * multidict=6.0.4=py38h5eee18b_0
> * ncurses=6.4=h6a678d5_0
> * ninja=1.11.0=h924138e_0
> * numpy=1.24.3=py38h14f4228_0
> * numpy-base=1.24.3=py38h31eccc5_0
> * oauthlib=3.2.2=py38h06a4308_0
> * openssl=3.0.15=h5eee18b_0
> * pip=24.2=py38h06a4308_0
> * protobuf=4.25.3=py38h12ddb61_0
> * pyasn1=0.4.8=pyhd3eb1b0_0
> * pyasn1-modules=0.2.8=py_0
> * pycparser=2.21=pyhd3eb1b0_0
> * pyjwt=2.9.0=pyhd8ed1ab_1
> * pyopenssl=24.2.1=py38h06a4308_0
> * pysocks=1.7.1=py38h06a4308_0
> * python=3.8.20=he870216_0
> * python_abi=3.8=2_cp38
> * pytorch=1.11.0=py3.8_cuda11.3_cudnn8.2.0_0
> * pytorch-mutex=1.0=cuda
> * pyyaml=6.0.2=py38h5eee18b_0
> * re2=2022.04.01=h295c915_0
> * readline=8.2=h5eee18b_0
> * requests=2.32.3=py38h06a4308_0
> * requests-oauthlib=2.0.0=py38h06a4308_0
> * rsa=4.7.2=pyhd3eb1b0_1
> * setuptools=75.1.0=py38h06a4308_0
> * six=1.16.0=pyh6c4a22f_0
> * sqlite=3.45.3=h5eee18b_0
> * tensorboard=2.6.0=py_0
> * tensorboard-plugin-wit=1.8.1=py38h06a4308_0
> * tk=8.6.14=h39e8969_0
> * typing-extensions=4.12.2=hd8ed1ab_0
> * typing_extensions=4.12.2=pyha770c72_0
> * urllib3=2.2.3=py38h06a4308_0
> * werkzeug=3.0.3=py38h06a4308_0
> * wheel=0.44.0=py38h06a4308_0
> * xz=5.4.6=h5eee18b_1
> * yaml=0.2.5=h7b6447c_0
> * yarl=1.11.0=py38h5eee18b_0
> * zipp=3.20.2=py38h06a4308_0
> * zlib=1.2.13=h5eee18b_1
> * pip:
>   
>   * einops==0.8.0
>   * joblib==1.4.2
>   * opencv-python==4.10.0.84
>   * pickle5==0.0.11
>   * scikit-learn==1.3.2
>   * scipy==1.10.1
>   * threadpoolctl==3.5.0
>   * tqdm==4.67.0
>     prefix: /home/ubuntu/anaconda3/envs/pedCMT
> 
> [results.txt](https://github.com/user-attachments/files/17760689/results.txt)
您好，我还有几个问题，希望您能解答我的疑问。
这是您的结果，
pie
Acc:93.1237721021611
F1:87.17948717948718
pre:85.61151079136691
Rec:88.80597014925374
AUC:91.7363184079602
matrix:[[355  20]
 [ 15 119]]
 
 这是我跑出来的结果
 pie
Acc: 0.8647058823529412
 f1: 0.7294117647058824
 precision_score: 0.7265625
 recall_score: 0.7322834645669292
 roc_auc_score: 0.8204498262782427
 confusion_matrix: [[348  35]
 [ 34  93]]

1：我想问一下，请问github上传的是最终版正确的代码吗，指标名字跟您的好像对不上，您的是pre，代码里是precision_score
2：请问是按照pie.py文件里默认参数配置训练模型的吗，把learn参数改为 False读取真实数据
3：您能把需要注意的点，相关训练命令告知我吗，万分感谢

期待您的回复，万分感谢！！！

**Comment by xbchen82** [comment]
1.	txt文件中的指标是写入txt文件中代码指明的字符串，得到的结果如“precision_score”等是使用print输出的指标。这一部分的代码是：
    filename = './pie_res.txt'
    with open(filename, 'a') as f:
        d_model = args.d_model
        dff = args.dff
        #batch_size = args.batch_size
        times_num = args.times_num
        layers = args.num_layers
        num_bnks = args.num_bnks
        bnks_layers = args.bnks_layers
        #lr = args.lr
        s = 'd_model: ' + str(d_model) + '\ndff: ' + str(dff) + '\nlayers: ' + str(layers) + '\ntimes_num: ' + str(times_num) + '\nbnks_layers: ' + str(bnks_layers) + '\nnum_bnks: ' + str(num_bnks) + '\n'
        #s = '\nbnks_layers: ' + str(bnks_layers) + '\nnum_bnks: ' + str(num_bnks) + '\n'
        #s = 'lr: ' + str(lr) +'\nnum_bnks: ' + str(num_bnks) + '\n'
        #st, ed = args.sta_f, args.end_f
        #s = 'start_frame :' + str(st) + '\n' + 'end_frame :' + str(ed) + '\n'
        f.write(s)
        accp = '\nAcc:' + str(acc * 100) + '---' + '\nF1:' + str(f1 * 100) + '---' + '\npre:' + str(pre_s * 100) + '---' + '\nRec:' + str(recall_s * 100) + '---' + '\nAUC:' + str(auc * 100) + '---'
        shift_line = '\n\n'
        f.write(accp)
        f.write(shift_line)

    print(f'Acc: {acc}\n f1: {f1}\n precision_score: {pre_s}\n recall_score: {recall_s}\n roc_auc_score: {auc}\n confusion_contrix: {contrix}')
2.	parser中设置的参数是默认的参数，与论文中实验的参数一致。也可以通过设置time_crop参数进行时间维度的随机裁剪。
3.	在函数main()中的data_opts参数，seq_type为crossing。当data_split_type为random时也会对结果产生影响。如：
    data_opts = {
        'fstride': 1,
        'sample_type': 'all',
        'height_rng': [0, float('inf')],
        'squarify_ratio': 0,
        'data_split_type': 'random',  # kfold, random, default
        'seq_type': 'crossing',  # crossing , intention
        'min_track_size': 15,  # discard tracks that are shorter
        'kfold_params': {'num_folds': 1, 'fold': 1},
        'random_params': {'ratios': [0.7, 0.15, 0.15],
                          'val_data': True,
                          'regen_data': False},
        'tte': [30, 60],
        'batch_size': 16
    }
建议你自己多试试，这个需要积累经验。

**Comment by abc11223301** [comment]
> 我用提供的代码，用JAAD数据集跑出来的结果，跟实际完全不符 结果如下 Acc: 0.6394557823129252 f1: 0.7800829875518672 precision_score: 0.6394557823129252 recall_score: 1.0 roc_auc_score: 0.5 confusion_matrix: [[ 0 53] [ 0 94]]
> 
> Acc: 0.18076923076923077 f1: 0.3061889250814332 precision_score: 0.18076923076923077 recall_score: 1.0 roc_auc_score: 0.5 confusion_matrix: [[ 0 426] [ 0 94]]
> 
> 这表明模型全都把类别只分成负类，模型没起效果

你好我也遇到了同样问题，请问你解决了吗

**Comment by AHNU-wang** [comment]
> > 我用提供的代码，用JAAD数据集跑出来的结果，跟实际完全不符 结果如下 Acc: 0.6394557823129252 f1: 0.7800829875518672 precision_score: 0.6394557823129252 recall_score: 1.0 roc_auc_score: 0.5 confusion_matrix: [[ 0 53] [ 0 94]]
> > Acc: 0.18076923076923077 f1: 0.3061889250814332 precision_score: 0.18076923076923077 recall_score: 1.0 roc_auc_score: 0.5 confusion_matrix: [[ 0 426] [ 0 94]]
> > 这表明模型全都把类别只分成负类，模型没起效果
> 
> 你好我也遇到了同样问题，请问你解决了吗
我也是，请问解决了吗

**Comment by ShifaTang** [comment]
> > 我用提供的代码，用JAAD数据集跑出来的结果，跟实际完全不符 结果如下 Acc: 0.6394557823129252 f1: 0.7800829875518672 precision_score: 0.6394557823129252 recall_score: 1.0 roc_auc_score: 0.5 confusion_matrix: [[ 0 53] [ 0 94]]
> > Acc: 0.18076923076923077 f1: 0.3061889250814332 precision_score: 0.18076923076923077 recall_score: 1.0 roc_auc_score: 0.5 confusion_matrix: [[ 0 426] [ 0 94]]
> > 这表明模型全都把类别只分成负类，模型没起效果
> 
> 你好我也遇到了同样问题，请问你解决了吗

我放弃了，没搞这个项目了
